### PR TITLE
create a single PopStateMachine instance per PopAwareForkResolutionComparator; pass the protected tree via the constructor as the protected and protecting trees are coupled anyway; minor cleanups

### PR DESCRIPTION
**Pull request by black0rwhite**

### CODE REVIEW

File: include/veriblock/pop/blockchain/pop/fork_resolution.hpp
Comment (Mr-Leshiy) [inline include/veriblock/pop/blockchain/pop/fork_resolution.hpp:447]: What was the intention to move ProtectedBlockTree& ed as an argument of the functions to the field of this struct ? What benefits these should bring to us ?

Comment (black0rwhite) [reply include/veriblock/pop/blockchain/pop/fork_resolution.hpp:447]: This is just a clean up, so the benefits are small. Basically, the states of the protected and protecting trees are coupled. Any pop state machine call on the protected tree changes the state of the protecting tree. So there's no way we could have 2 protected trees per 1 protecting tree. This change leads to a minor simplification of code.

**Comment by Mr-Leshiy** [comment]
LGTM, after update to the latest develop we can merge it

**Review by Mr-Leshiy — APPROVED** [review]

**Pull request merged**
